### PR TITLE
dashboard: remove rgw api host,port,scheme

### PR DESCRIPTION
**Pull request by dsavineau**
We don't need to have dedicated variables for the RGW integration into
the Ceph Dashboard and need to be manually filled.
Instead we can use the current values from the RGW nodes by using the
IP and port from the first RGW instance of the first RGW node via the
radosgw_address and radosgw_frontend_port variables.
We don't need to specify all RGW nodes, this will be done automatically
with one node.
The RGW api scheme is using the radosgw_frontend_ssl_certificate variable
to determine if the value is http or https. This variable is also reuse
as a condition for the ssl verify task.

Signed-off-by: Dimitri Savineau <dsavinea@redhat.com>

### CODE REVIEW

File: roles/ceph-dashboard/tasks/configure_dashboard.yml [hunk 1]
```diff
@@ -153,25 +153,22 @@
       changed_when: false
 
     - name: set the rgw host
-      command: "{{ container_exec_cmd }} ceph dashboard set-rgw-api-host {{ dashboard_rgw_api_host }}"
+      command: "{{ container_exec_cmd }} ceph dashboard set-rgw-api-host {{ hostvars[groups[rgw_group_name][0]]['rgw_instances'][0]['radosgw_address'] }}"
```
[the remark this answers is not in the document]
Comment (dsavineau) [reply roles/ceph-dashboard/tasks/configure_dashboard.yml]: it's already in a block when there's rgw nodes defined in the inventory

https://github.com/ceph/ceph-ansible/blob/master/roles/ceph-dashboard/tasks/configure_dashboard.yml#L121-L123

File: roles/ceph-dashboard/tasks/configure_dashboard.yml [hunk 2]
```diff
@@ -153,25 +153,22 @@
...
       changed_when: false
       delegate_to: "{{ groups[mon_group_name][0] }}"
       run_once: true
-      when: dashboard_rgw_api_host != ''
 
     - name: set the rgw port
-      command: "{{ container_exec_cmd }} ceph dashboard set-rgw-api-port {{ dashboard_rgw_api_port }}"
+      command: "{{ container_exec_cmd }} ceph dashboard set-rgw-api-port {{ hostvars[groups[rgw_group_name][0]]['rgw_instances'][0]['radosgw_frontend_port'] }}"
```
[the remark this answers is not in the document]
Comment (dsavineau) [reply roles/ceph-dashboard/tasks/configure_dashboard.yml]: see my previous comment

File: roles/ceph-dashboard/tasks/configure_dashboard.yml [hunk 3]
```diff
@@ -185,7 +182,9 @@
       changed_when: false
       delegate_to: "{{ groups[mon_group_name][0] }}"
       run_once: true
-      when: dashboard_rgw_api_no_ssl_verify | bool
+      when:
+        - dashboard_rgw_api_no_ssl_verify | bool
+        - radosgw_frontend_ssl_certificate | length > 0
```
[the remark this answers is not in the document]
Comment (dsavineau) [reply roles/ceph-dashboard/tasks/configure_dashboard.yml]: radosgw_frontend_ssl_certificate is a variable defining the the ssl certificate path. This value is by default set to an empty string (so length == 0).
The ssl verify is only disable when we provdie the ssl certificate (ie rgw is using https) and the dashboard_rgw_api_no_ssl_verify flag is set to true.

[the remark this answers is not in the document]
Comment (b-ranto) [reply roles/ceph-dashboard/tasks/configure_dashboard.yml]: Ah, ok, so you can look at the other condition as 'we are using https for rgw'. It makes sense to use it there, then.

File: roles/ceph-dashboard/tasks/configure_dashboard.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (b-ranto) [reply roles/ceph-dashboard/tasks/configure_dashboard.yml]: All good then.

**Pull request merged**